lib/include/up-client-zenoh-cpp/rpc/zenohRpcClient.h

```diff
@@ -25,70 +25,120 @@
+
+    class ZenohRpcClient : public uprotocol::rpc::RpcClient {
+
+        public:
+
+            ZenohRpcClient(const ZenohRpcClient&) = delete;
+            ZenohRpcClient& operator=(const ZenohRpcClient&) = delete;
+
+            /**
+            * The API provides an instance of the zenoh RPC client
+            * @return instance of ZenohUTransport
+            */
+            static ZenohRpcClient& instance(void) noexcept;
+
+            /**
+            * init the zenohRpcClient 
+            * @return Returns OK on SUCCESS and ERROR on failure
+            */
+            uprotocol::v1::UStatus init(ZenohRpcClientConfig *config = nullptr) noexcept;
+
+            /**
+            * Terminates the zenoh RPC client  - the API should be called by any class that called init
+            * @return Returns OK on SUCCESS and ERROR on failure
+            */
+            uprotocol::v1::UStatus term() noexcept; 
+
+            /**
+            * API for clients to invoke a method (send an RPC request) and receive a future that will hold the response 
+            * Client will set method to be the URI of the method they want to invoke, 
+            * payload to the request message, and attributes with the various metadata for the 
+            * method invocation.
+            * @param topic The method URI to be invoked (MICRO FORM)
+            * @param payload The request message to be sent to the server.
+            * @param options RPC method invocation call options, see {@link CallOptions}
+            * @return Returns the future 
+            */
+            std::future<uprotocol::rpc::RpcResponse> invokeMethod(const uprotocol::v1::UUri &topic, 
+                                                                  const uprotocol::utransport::UPayload &payload, 
+                                                                  const uprotocol::v1::CallOptions &options) noexcept;
+
+            /**
+            * API for clients to invoke a method (send an RPC request) , the response will be received through the callback
+            * Client will set method to be the URI of the method they want to invoke, 
+            * payload to the request message, and attributes with the various metadata for the 
+            * method invocation.
+            * @param topic The method URI to be invoked (MICRO FORM)
+            * @param payload The request message to be sent to the server.
+            * @param options RPC method invocation call options, see {@link CallOptions}
+            * @param user callback
+            * @return Returns the UStatus 
+            */
+            uprotocol::v1::UStatus invokeMethod(const uprotocol::v1::UUri &topic,
+                                                const uprotocol::utransport::UPayload &payload,
+                                                const uprotocol::v1::CallOptions &options,
+                                                const uprotocol::utransport::UListener &callback) noexcept;
```

Comment:
This would need to be updated if UListener is changed to replace functors with lambdas. See [my comment](https://github.com/eclipse-uprotocol/up-cpp/pull/68#discussion_r1532755486) on the up-cpp PR

Reply:
Not an issue for now

Reply:
@MelamudMichael if something is not going to be resolved, add a link to the issue that will be used to track the change.